lib/compute/agent-nodes.ts

```diff
@@ -0,0 +1,185 @@
+    def security_group = "${stackConfig.agent_node_security_group}"
+    def subnet_id = "${stackConfig.subnet_id}"
+    def instance_profile_arn = "${this.AgentNodeInstanceProfileArn}"
+    def sshKeysASMSecretId = "${this.SSHEC2KeySecretId}"
+    def ec2_tags = [new EC2Tag('Name', 'jenkins-agent-node')]
+
+    def agent_node_template = new SlaveTemplate(
+      // String ami
+      ami_id,
+      // String zone
+      '',
+      // SpotConfiguration spotConfig
+      null,
+      // String securityGroups
+      security_group,
+      // String remoteFS
+      '/var/jenkins',
+      // InstanceType type
+      InstanceType.fromValue(instance_type),
+      // boolean ebsOptimized
+      false,
+      // String labelString
+      worker_label_string,
+      // Node.Mode mode
+      Node.Mode.NORMAL,
+      // String description
+      'Jenkins Agent Node',
+      // String initScript
+      init_script,
+      // String tmpDir
+      '',
+      // String userData
+      '',
+      // String numExecutors
+      number_of_executors,
+      // String remoteAdmin
+      remote_user,
+      // AMITypeData amiType
+      new UnixData(null, null, null, null),
+      // String jvmopts
+      '',
+      // boolean stopOnTerminate
+      false,
+      // String subnetId
+      subnet_id,
+      // List<EC2Tag> tags
+      ec2_tags,
+      // String idleTerminationMinutes
+      '30',
+      //int minimumNumberOfInstances,
+      1,
+      //int minimumNumberOfSpareInstances,
+      1,
+      // String instanceCapStr
+      '20',
+      // String iamInstanceProfile
+      instance_profile_arn,
+      //  boolean deleteRootOnTermination
+      true,
+      // boolean useEphemeralDevices
+      false,
+      // String launchTimeoutStr
+      '1800',
+      // boolean associatePublicIp
+      false,
+      // String customDeviceMapping
+      '/dev/xvda=:100:true:::encrypted',
+      // boolean connectBySSHProcess
+      false,
+      // boolean monitoring,
+      true,
+      // boolean t2Unlimited,
+      false,
+      // ConnectionStrategy connectionStrategy,
+      null,
+      // int maxTotalUses,
+      -1,
+      // List<? extends NodeProperty<?>> nodeProperties,
+      null,
+      // HostKeyVerificationStrategyEnum hostKeyVerificationStrategy,
+      HostKeyVerificationStrategyEnum.OFF,
+      //Tenancy tenancy,
+      Tenancy.Default,
+      //EbsEncryptRootVolume ebsEncryptRootVolume,
+      EbsEncryptRootVolume.ENCRYPTED,
+    )
+
+    def new_cloud = new AmazonEC2Cloud(
+      // String cloudName
+      ec2_cloud_name,
+      // boolean useInstanceProfileForCredentials
+      true,
+      // String credentialsId
+      '',
+      // String region
+      region,
+      // String privateKey
+      null,
+      // String sshKeysCredentialsId
+      sshKeysASMSecretId,
+      // String instanceCapStr
+      "10",
+      // List<? extends SlaveTemplate> templates
+      [agent_node_template],
+      // String roleArn
```

Comment:
Is this the role of the running agent node? I think we should have one so we can use SSM to ssh into the host just like the [main node](https://github.com/opensearch-project/opensearch-ci/blob/86aea2e78a29cd02e1df77031a1b0b57e09f2dd9/lib/compute/jenkins-main-node.ts#L70) for troubleshooting.

Reply:
- it looks like the instance profile has a role, maybe a comment would be useful instead

Reply:
Added